### PR TITLE
backend: Add create commit method for GitHub

### PR DESCRIPTION
**Pull request by septum**
### Description

- Added CreateCommit method in GitHub service

### Testing Performed

Manual

### CODE REVIEW

File: backend/service/github/github.go
Comment (jdslaugh) [inline backend/service/github/github.go:626]: Most of this functionality is identical to the CreateBranch function, so I think we can pull out the similar pieces into a helper function and utilize it in both places so we don't have duplicate logic.

**Review by jdslaugh — CHANGES_REQUESTED** [review]

File: backend/service/github/github.go
Comment (septum) [reply backend/service/github/github.go:626]: Sounds good! I'll wait until we merge the changes in #3106 to encapsulate this logic.

**Review by jdslaugh — APPROVED** [review]

**Pull request merged**
